python/simpler/worker.py

```
@@ -4305,6 +4305,52 @@ def _on_cancel(_signum, _frame):
+        self.nbytes = int(nbytes)
+        self.base = worker.alloc_pinned_host(self.nbytes)
+
+    def __del__(self) -> None:
+        base = self.base
+        if base == 0:
+            return
+        self.base = 0
+        try:
+            self._worker.free_pinned_host(base)
+        except Exception:  # noqa: BLE001 -- ChipWorker.finalize already reclaims every still-live allocation
+            pass
```

Ruff (0.16.1)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)